kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -126,6 +131,67 @@ private KaldbServiceGrpc.KaldbServiceFutureStub getKaldbServiceGrpcClient(String
         .withCompression("gzip");
   }
 
+  @VisibleForTesting
+  public static Collection<String> getSnapshotsToSearch(
+      SnapshotMetadataStore snapshotMetadataStore,
+      SearchMetadataStore searchMetadataStore,
+      long startTimeMs,
+      long endTimeMs) {
+    // step 1 - find all snapshots that match time window
+    Set<String> snapshotsToSearch =
+        snapshotMetadataStore
+            .getCached()
+            .stream()
+            .filter(
+                snapshotMetadata ->
+                    containsDataInTimeRange(
+                        snapshotMetadata.startTimeEpochMs,
+                        snapshotMetadata.endTimeEpochMs,
+                        startTimeMs,
+                        endTimeMs))
+            .map(snapshotMetadata -> snapshotMetadata.name)
+            .collect(Collectors.toSet());
+
+    // TODO: in the future, we'd have another field in search metadata to indicate if it's hosted by
```

Comment:
SearchMetada doesn't have an `isLive` today

We can perhaps infer from `snapshotName` since it will start with `LIVE_something`
```
  public final String snapshotName;
  public final String url;
```

But Bryan and I were thinking SearchMetada in the future could have another field which represents am I a cache node, am I an indexer node. And even associate a query priority score to the node so that it's not "random" but we would prefer one over the other in a more nuanced way

let me know if you think we should do `live` vs non live for now?